trade-finance-manager-ui/server/controllers/utilisation-reports/helpers/reconciliation-for-report-helper.test.ts

```diff
@@ -411,9 +410,7 @@ describe('reconciliation-for-report-helper', () => {
 
       const feeRecordPaymentGroups: FeeRecordPaymentGroup[] = [{ ...aFeeRecordPaymentGroup(), feeRecords: [feeRecord], status }];
 
-      const checkedCheckboxId = `feeRecordIds-${feeRecordId}-reportedPaymentsCurrency-${feeRecordReportedPaymentsCurrency}-status-${status}`;
-
-      const isCheckboxChecked = (checkboxId: PremiumPaymentsTableCheckboxId) => checkboxId === checkedCheckboxId;
+      const isCheckboxChecked = (feeRecordIdsToCheck: number[]) => feeRecordIdsToCheck.every((feeRecordIdToCheck) => feeRecordIdToCheck === feeRecordId);
```

Comment:
I think this function is overly complex for a test function in that it looks like it is trying to imitate the real implementation. We could do two things:
```
const isCheckboxChecked = (ids: number[]) => ids === [feeRecordId];
```
Or I'd be interested to know if something like this works, but perhaps unnecessarily complex:
```
const isCheckboxChecked = jest.fn().mockReturnValue(false);
when(isCheckboxChecked).calledWith([feeRecordId]).mockReturnValue(true);
```

Reply:
I think the first would fail as objects are compared by reference, implemented the second approach thanks

Reply:
Ahh yes of course!